pyexchange/uniswapv2_analytics.py

```diff
@@ -267,9 +274,45 @@ def get_our_mint_txs(self, pair_address: Address) -> dict:
 
         result = self.graph_client.query_request(get_our_mint_txs_query, variables)['mints']
 
-        sorted_mints = sorted(result, key=lambda mint: mint['timestamp'], reverse=True)
+        sorted_mints = sorted(result, key=lambda mint: mint['transaction']['blockNumber'], reverse=True)
         return sorted_mints
 
+    def get_block_trade(self, pair_address: Address, block: int) -> List:
+        """
+            Retrieve pair data for a given block
+        """
+        assert (isinstance(pair_address, Address))
+        assert (isinstance(block, int))
+
+        get_pair_data_query = """query ($id: Bytes!, $block: Int!)
+            {
+                pairs (block: {number: $block}, where: {id: $id}) {
+                    totalSupply
+                    token0 {
+                        id
+                    }
+                    token1 {
+                        id
+                    }
+                    token0Price
+                    token1Price
+                    volumeToken0
+                    volumeToken1
+                    reserve0
+                    reserve1
+                    txCount
+                    id
+                }
+            }
+        """
```

Comment:
Questions about the query:
1. Do we still need to query `volumeToken0` and `volumeToken1`?
2. Where can I find docs for the `pairs` query?  I looked [here](https://thegraph.com/explorer/subgraph/graphprotocol/uniswap) but didn't find these parameters.

Reply:
1. Nope, I've removed those attributes.
2. Each version of Uniswap has its own subgraph, v2 is available [here](https://thegraph.com/explorer/subgraph/uniswap/uniswap-v2). Block is available as a filter on all the top level uniswapv2 subgraph entities.

Reply:
I'm curious what `reserve0` and `reserve1` show.  The documentation didn't seem to offer a description of these fields.

Reply:
reserve0 and reserve1 are reflection of the pair contract (https://github.com/Uniswap/uniswap-v2-subgraph/blob/master/schema.graphql#L64) (https://uniswap.org/docs/v2/smart-contracts/pair/#getreserves), and represent the the amount of each asset for a given pair locked in the contract.